TLM/TLM/Util/Extensions/NetSegmentExtensions.cs

```diff
@@ -283,63 +284,69 @@ public static class NetSegmentExtensions {
             bool reverse = false,
             bool sort = true) {
             // TODO refactor together with getSegmentNumVehicleLanes, especially the vehicle type and lane type checks
-            var laneList = new List<LanePos>();
 
             NetInfo segmentInfo = netSegment.Info;
 
-            if (segmentInfo?.m_lanes == null)
-                return laneList;
+            byte maxLanes = (byte)(segmentInfo?.m_lanes?.Length ?? 0);
+            if (maxLanes == 0)
+                return new List<LanePos>(0);
```

Comment:
Returning a null or some special indication of no value might save allocation of an empty array. If this does not happen too often, disregard this comment.

Reply:
I can create `private static readonly` empty list to return in these cases.

_While returning `null` would be even better, it would potentially require lot of code changes and might impact on asset editor support and Move It integration (I don't know enough about those features yet)._

Reply:
No performance critical part of the code should call this function to begin with. maybe a comment.